### PR TITLE
Being shot into space by the mass driver now reconnects you to Terry - Take 2

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
To see the original PR, see #57405. See original PR description below.

> The mass driver doesn't see enough use in my opinion, despite being very cool and flavorful, and I think a lot of this has to do with how uncompetitive it is compared to most other forms of body disposal. Not only does it require you taking the body to a part of the map right next to a much more thorough disposal method (the incinerator), it has a nasty habit of shooting bodies that bounce off something in space and float right back, or even loop around and hit the station on the other side.
>
> To make the mass driver more competitive, being launched off the station z-level by one will now subject you to a fate infinitely worse than being shot into the vast emptiness of space: it forcibly disconnects you from your current server and reconnects you to Terry instead. Once there, you're immediately spawned on the fringe of the map from the direction you were fired from, and launched towards the station. In the unfortunate event that you were for some reason already connected to Terry, you will instead be sent to the empty void that is Campbell.
>
> I believe this will give the mass driver a special niche in body disposal. While other methods like gibbing, incinerating, or simply debraining still have their advantages when it comes to quick and discreet disposal, none of them actively inflict pain and distress on the other player like the mass driver can.

[Here's a link to a test video, this doesn't show actual server hopping, but this is basically what it'd look like (minus the loading time)](https://cdn.discordapp.com/attachments/326831214667235328/1020515976669827072/terry_cannon.mp4)

## Why It's Good For The Game
Gives the mass driver some teeth, adds a new way to really show someone how little you respect them, finally makes good on a 2  year promise to finish this.

## Changelog

:cl: Ryll/Shaps
add: Being shot into space via mass driver will now send you to hell
/:cl:

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
There's no proof that icebox is in any server's rotation so I don't see that being an issue

**Comment by Shadowflame909** [comment]
me when i send the changeling to terry (they will never know)

Sec should keep an eye out to shoot and kill anything that was evil enough/pitiful enough to get shot out of a mass driver

**Comment by Striders13** [comment]
make it connect you to Campbell instead.

**Comment by Mothblocks** [comment]
you couldve just unlocked the comments on the other pr you know

File: code/controllers/configuration/entries/comms.dm [hunk 1]
```diff
@@ -25,3 +25,9 @@
 
 /datum/config_entry/string/cross_comms_network
 	protection = CONFIG_ENTRY_LOCKED
+
+/datum/config_entry/string/hell
+	default = "byond:\\terry.tgstation13.org:3336"
+
+/datum/config_entry/string/the_abyss
+	default = "byond:\\campbell.tgstation13.org:6337"
```
Comment (Mothblocks) [inline code/controllers/configuration/entries/comms.dm:33]: //

File: code/datums/refugee.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	message_admins("[ADMIN_LOOKUPFLW(new_body)] has been shot towards this station from another server.")
+	new_body.status_flags &= GODMODE
+	addtimer(CALLBACK(src, .proc/delayed_spawn, new_body, our_new_player), 4 SECONDS) // delay actually spawning them and flinging them at the station a bit in case they're loading so they can see it
+
+/// fling 'em
+/datum/refugee/proc/delayed_spawn(mob/living/carbon/human/the_body, mob/dead/new_player/our_new_player)
+	our_new_player.transfer_character()
+	var/turf/pickedstart
+	var/turf/pickedgoal
+	var/max_i = 10//number of tries to spawn the incoming body
```
Comment (SuperNovaa41) [inline code/datums/refugee.dm:32]: shouldnt this just be a define?

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

**Comment by AdipemDragon** [comment]
how.... good is it at preserving the exact mob that was thrown out?

**Comment by moocowswag** [comment]
If it preserves a mob at all it means adminbus can leak into other rounds.

**Comment by VioletN** [comment]
This is hilarious, but there should probably be a config option to turn it on/off

**Comment by Ryll-Ryll** [comment]
Speaking as both a maintainer and gamemaster I think we should be able to trust our players to use this content responsibly and not arbitrarily limit them

**Comment by Colovorat** [comment]
wait does it transfer all the shit with you too now??

**Comment by Ryll-Ryll** [comment]
No that arbitrary limit is retained

**Comment by JohnFulpWillard** [comment]
this is one way to get pop to campbell

File: code/controllers/configuration/entries/comms.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/controllers/configuration/entries/comms.dm:33]: I just followed the format used elsewhere for server addresses (see a few lines above this, line 22). I'll change it if you want tho

File: code/datums/components/exile.dm [hunk 3]
```diff
@@ -1,3 +1,8 @@
+/// euros
+#define HELL "byond:\\terry.tgstation13.org:3336"
```
Comment (Mothblocks) [inline code/datums/components/exile.dm:2]: youre doing backslashes again

**Comment by theta060** [comment]
finally i can launch suicide bombers at terry for shitposting on all stations

File: code/datums/refugee.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+/datum/refugee/New(incoming_ckey, incoming_name, heading_dir)
+	. = ..()
+	expected_ckey = incoming_ckey
+	char_name = incoming_name
+	launching_dir = heading_dir
+
+/// make the body once they're logged in
+/datum/refugee/proc/execute_introduction(mob/dead/new_player/our_new_player)
+	GLOB.refugees -= src
+	var/turf/random_standby = pick(GLOB.blobstart)
```
Comment (ShizCalev) [inline code/datums/refugee.dm:19]: GLOB.blobstart can be empty. Either have a backup or have it crash if null.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/refugee.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	var/mob/living/carbon/human/new_body = our_new_player.create_character(random_standby)
+	new_body.real_name = char_name
```
Comment (ShizCalev) [inline code/datums/refugee.dm:21]: new_body.fully_replace_character_name(new_body.real_name, char_name)

File: code/datums/refugee.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	var/turf/pickedstart
+	var/turf/pickedgoal
+	var/max_i = 10//number of tries to spawn the incoming body
+	var/startSide
+
+	for(var/i in 1 to max_i)
+		startSide = turn(launching_dir, 180)
+		var/startZ = pick(SSmapping.levels_by_trait(ZTRAIT_STATION))
+		pickedstart = spaceDebrisStartLoc(startSide, startZ)
+		pickedgoal = pick(GLOB.blobstart) // just chuck em somewhere on station
```
Comment (ShizCalev) [inline code/datums/refugee.dm:39]: blob here too.

File: code/game/machinery/mass_driver.dm [hunk 7]
```diff
@@ -44,7 +47,10 @@
 			if(O_limit >= 20)
 				audible_message(span_notice("[src] lets out a screech, it doesn't seem to be able to handle the load."))
 				break
-			use_power(active_power_usage)
```
Comment (ShizCalev) [inline code/game/machinery/mass_driver.dm:47]: This doesn't look intended

**Comment by Ryll-Ryll** [comment]
will maybe finish later idk

**Pull request closed without merging**